### PR TITLE
feat(executions): Adding redirect for execution without application

### PR DESCRIPTION
**Pull request by alanmquach**
Redirects from `/#/executions/ABC123` to `/#/applications/egg/executions/details/ABC123`.

### CODE REVIEW

File: app/scripts/modules/core/src/pipeline/pipeline.states.ts [hunk 1]
```diff
@@ -92,5 +93,18 @@ module(PIPELINE_STATES, [APPLICATION_STATE_PROVIDER]).config([
     };
 
     applicationStateProvider.addChildState(pipelines);
+
+    const executionsLookup: INestedState = {
+      name: 'executionLookup',
+      url: '/executions/:executionid',
+      params: {
+        executionid: { dynamic: true },
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/pipeline/pipeline.states.ts]: why not camel case this?

Comment (alanmquach) [reply app/scripts/modules/core/src/pipeline/pipeline.states.ts]: I feel like there was some unspoken rule from the ancient nodejs days where path params were all lowercase?

In other words, idk so let me just camel case it.

**Comment by christopherthielen** [comment]
I would prefer to see this as a [`redirectTo` on the state definition](https://ui-router.github.io/core/docs/latest/interfaces/state.statedeclaration.html#redirectto), and no ui component doing the work.

File: app/scripts/modules/core/src/pipeline/pipeline.states.ts [hunk 2]
```diff
@@ -92,5 +94,39 @@ module(PIPELINE_STATES, [APPLICATION_STATE_PROVIDER]).config([
...
+      },
+      redirectTo: transition => {
+        const { executionId } = transition.params();
+        const executionService: ExecutionService = transition.injector().get('executionService');
+
+        if (!executionId) {
+          return undefined;
+        }
+
+        return Promise.resolve()
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/pipeline/pipeline.states.ts]: can this `Promise.resolve()` be removed?

```js
        return executionService.getExecution(executionId)
          .then(execution =>
```

Comment (alanmquach) [reply app/scripts/modules/core/src/pipeline/pipeline.states.ts]: `redirectTo` expects a `Promise` so needed to convert the `IPromise` from `executionService`.

Is this the right way to do that or is there a better way?

**Pull request merged**